app/frontend/actions.js

```diff
@@ -60,13 +63,50 @@ module.exports = ({setState, getState}) => {
     )
   }
 
+  const handleError = (errorName, e, options) => {
+    if (e && e.name) {
+      // is a error
+      debugLog(e)
+      captureBySentry(e)
+      setState({
+        [errorName]: {
+          code: e.name,
+          params: {
+            message: e.message,
+            ...options,
+          },
+        },
+        error: e,
+      })
+    } else {
+      setState({
+        [errorName]: e,
+      })
```

Comment:
why does this function do anything with state? From what I see, you extracted some state-setting from calling functions and moved them here.

Reply:
I dont really see a problem with this function changing state, after all all the functions in `actions.js` change state. Yes I extracted some state changing functionality from calling function and moved it here, I see several advantages to this solution:

1. This function ensures that setting the error state is handled the same way every time an error occurs and its handling is defined in one place.

2. This way, setting an error state is dedicated from setting other states, in lot of cases states that has nothing to do with the fact that an error has occurred.

3. The extracted code was in lot of cases redundant since I tried to make the error handling as consistent as possible. This function spared us tens of redundant lines.

4. The function supports the way I implemented the sentry error capturing, since every error is handled by this function, every error checked against translations whether the error is expected or not, instead of calling `captureBySentry` all over the code.

What do you think?